src/jou_compiler.h

```diff
@@ -370,7 +375,8 @@ Type *create_struct(
 void free_type(Type *type);
 
 bool is_integer_type(const Type *t);  // includes signed and unsigned
-bool is_number_type(const Type *t);  // integers, floats, doubles
+bool is_number_type(const Type *t);  // integers, doubles
+bool is_float_type(const Type *t); // floats need split with doubles
```

Comment:
What's going on here? It doesn't make any sense to me. Floats are numbers, so `is_number_type()` should check for integers, floats and doubles, as the original comment said.

Reply:
But in  file "typecheck.c" line 268 ~ 272, how to we identify float or double?
```c
    if (got_numbers && ???) { // What to check?
        cast_type = floatType;
    }
    if (got_numbers && !got_integers) {
        cast_type = doubleType;
    }
```

Reply:
if not got_integers is double, so how do we split up double/float?

Reply:
As I mentioned in another comment, I think you should use `double` whenever the left side or the right side is a double.

```c
if (got_numbers && !got_integers) {
    cast_type = (lhstypes->type == doubleType || rhstypes->type == doubleType) ? doubleType : floatType;
}
```